Sources/Lexical.swift

```diff
@@ -26,6 +26,23 @@ public func letter(_ source: [Character]) -> (Character, [Character])? {
   }
 }
 
+public func word(_ value: String) -> ([Character]) -> (String, [Character])? {
+  let wordParser = letter+ ^^ { (cs: [Character]) -> String in
+    print(cs)
+    return String(cs)
+  }
+
+  return { source in
+    return wordParser(source).flatMap { word in
```

Comment:
* Remove `return`
* The filtering operation could be better expressed through `Optional.filter` (which technically doesn't exist, and thus needs to be implemented as an extension of `Optional`).